s/start.go

```diff
@@ -15,43 +15,66 @@ import (
 	"github.com/capatazlib/go-capataz/c"
 )
 
-// monitorLoop does the initialization of supervisor's children and then runs an
-// infinite loop that monitors each child error.
-func (sup Supervisor) monitorLoop(
+// runMonitorLoop does the initialization of supervisor's children and then runs
+// an infinite loop that monitors each child error.
+//
+// This function is used for both async and sync strategies, given this, we
+// receive an onStart and onTerminate callbacks that behave differently
+// depending on which strategy is used:
+//
+// 1) When called with the async strategy, these callbacks will interact with
+// gochans that communicate with the spawner goroutine.
+//
+// 2) When called with the sync strategy, these callbacks will return the given
+// error, note this implementation returns the result of the callback calls
+//
+func runMonitorLoop(
 	ctx context.Context,
-	startCh chan startError,
-	terminateCh chan terminateError,
+	spec SupervisorSpec,
+	runtimeName string,
 	notifyCh chan c.ChildNotification,
-) {
-	defer close(terminateCh)
-
+	onStart c.NotifyStartFn,
+	onTerminate func(terminateError),
```

Comment:
Curious, why not a type for onTerminate?

Reply:
Yes, good idea, I'll do this for consistency.